### DISCUSSION TITLE
[Feature Request][Trace Viewer] Support span references / links

### DISCUSSION THREAD

**Discussion by erin-noe-payne** [grafana/grafana#41294] [Observability]
Trace spans can include links to other spans in other traces or in the same trace  ([otel spec](https://github.com/open-telemetry/opentelemetry-specification/blob/ef6e2f41a905a8c9b89d9e0895d9648ee7645f65/specification/overview.md#links-between-spans)). I should be able to visualize these links from the trace viewer, and to follow them to the linked trace & span.

There is already a draft PR https://github.com/grafana/grafana/pull/39630 starting an implementation, but it does not seem to have gotten traction so far.

# UX
- From the trace timeline view, I would like some visual indicator that a span has links associated with it
- From the span detail view, I should be able to view each link, any attributes associated with the link, and I should be able to click a hyperlink to view the linked trace & ideally the span (related https://github.com/grafana/grafana/discussions/40720)

See Jaeger's UI as a reference

![span links](https://user-images.githubusercontent.com/918178/140187631-5735cb4b-f994-4ed2-b5a8-87b60414181a.png)

**Comment by erin-noe-payne** [comment]
I'm starting a draft PR here: https://github.com/grafana/grafana/pull/41438

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
